packages/core/src/lib/template-engine/built-in-extensions/masking/maskingTagRunner.ts

```diff
@@ -0,0 +1,31 @@
+import * as nunjucks from 'nunjucks';
+import {
+  TagRunner,
+  TagRunnerOptions,
+  VulcanInternalExtension,
+} from '@vulcan-sql/core/models';
+
+@VulcanInternalExtension()
+export class MaskingTagRunner extends TagRunner {
+  public tags = ['masking'];
+
+  public async run({ args }: TagRunnerOptions) {
+    const columnName = args[0];
+    const prefix = args[1];
+    const padding = `'${args[2]}'`;
+    const suffix = args[3];
+
+    const prefixSql = `substr(${columnName}, 1, ${prefix})`;
+    if (suffix === 0) {
+      return new nunjucks.runtime.SafeString(
+        `concat(${prefixSql}, ${padding})`
+      );
+    }
+
+    // Consider the currently supported connector, using a general function to implement it.
+    const pos = Number(suffix) - 1;
+    return new nunjucks.runtime.SafeString(
+      `concat(${prefixSql}, ${padding}, substr(${columnName}, length(${columnName}) - ${pos}, ${suffix}))`
+    );
```

Comment:
We should handle the strings which are too short, for example:

|Data|Masking function|Result|
|------|-------|--------|
|freda|partial(1,'xxx', 1)|fxxxa|
|freda|partial(2,'xxx',2)|frxxxda|
|freda|partial(2,'xxx',3)|xxx|